docs/architecture/adr-006-light-client-refactor.md

```diff
@@ -0,0 +1,214 @@
+component user.
+
+```rust
+pub struct Handle {
+    sender: channel::Sender<Event>,
+}
+...
+impl Handle {
+    pub fn verify_to_target(&mut self, height: Height) -> Result<Header, &'static str> {
+        let (sender, receiver) = channel::bounded::<Event>(1);
+        let callback = Callback::new(move |result| {
+            let event = match result {
+                Ok(header) => {
+                    Event::Verified(header)
+                },
+                Err(err) => {
+                    Event::FailedVerification()
+                }
+            };
+            sender.send(event).unwrap();
+        });
+
+        self.sender.send(Event::VerifyToTarget(height, callback)).unwrap();
+
+        match receiver.recv().unwrap() {
+            Event::Verified(header) => {
+                return Ok(header);
+            },
+            Event::FailedVerification() => {
+                return Err("too bar");
+            },
+            _ => {
+                return Err("that was unexpected");
+            }
+
+        }
+    }
+}
+
+...
+// Supervisor
+pub struct Supervisor {
+    ...
+    sender: channel::Sender<Event>,
+    receiver: channel::Receiver<Event>,
+}
+
+impl Supervisor {
+    ...
+    pub fn handle(&mut self) -> Handle {
+        let sender = self.sender.clone();
+
+        return Handle::new(sender);
+    }
+    ...
+    pub fn run(mut self) {
+        thread::spawn(move || {
+            loop {
+                let event = self.receiver.recv().unwrap();
+                match event {
+                    Event::Terminate(sender) => {
+                        println!("Terminating light client");
+                        sender.send(()).unwrap();
+                        return
+                    },
+                    Event::VerifyToTarget(height, callback) => {
+                        let outcome = self.verify_to_target(height);
+                        callback.call(outcome);
+                    },
+                    _ => {
+                        // NoOp?
+                    },
+                }
+            }
+        });
+    }
+}
+
+```
+## Status
+
+Implemented:
+    * [Verification Predicates](https://github.com/informalsystems/tendermint-rs/blob/e2335c4/light-client/src/predicates.rs)
+    * [Verifier Component](https://github.com/informalsystems/tendermint-rs/blob/e2335c4/light-client/src/components/verifier.rs)
+    * [IO Component](https://github.com/informalsystems/tendermint-rs/blob/e2335c4/light-client/src/components/io.rs)
+    * [Scheduler Component](https://github.com/informalsystems/tendermint-rs/blob/e2335c4/light-client/src/components/scheduler.rs)
+    * [Supervisor](https://github.com/informalsystems/tendermint-rs/blob/e2335c40b1c5e1f7d47ee28ae5f9cc679730b7a2/light-client/src/supervisor.rs)
+    * [Example](https://github.com/informalsystems/tendermint-rs/blob/e2335c40b1c5e1f7d47ee28ae5f9cc679730b7a2/light-client/examples/light_client.rs)
```

Comment:
```suggestion

* [Verification Predicates](https://github.com/informalsystems/tendermint-rs/blob/e2335c4/light-client/src/predicates.rs)
* [Verifier Component](https://github.com/informalsystems/tendermint-rs/blob/e2335c4/light-client/src/components/verifier.rs)
* [IO Component](https://github.com/informalsystems/tendermint-rs/blob/e2335c4/light-client/src/components/io.rs)
* [Scheduler Component](https://github.com/informalsystems/tendermint-rs/blob/e2335c4/light-client/src/components/scheduler.rs)
    * [Supervisor](https://github.com/informalsystems/tendermint-rs/blob/e2335c40b1c5e1f7d47ee28ae5f9cc679730b7a2/light-client/src/supervisor.rs)
    * [Example](https://github.com/informalsystems/tendermint-rs/blob/e2335c40b1c5e1f7d47ee28ae5f9cc679730b7a2/light-client/examples/light_client.rs)
```

This will fix the render of the markdown.